src/postfork.cpp

```diff
@@ -158,8 +158,6 @@ bool set_child_group(job_t *j, pid_t child_pid) {
 }
 
 bool maybe_assign_terminal(const job_t *j) {
-    assert(j->pgid > 1 && "maybe_assign_terminal() called on job with invalid pgid!");
```

Comment:
(We could also keep a j->pgid != INVALID_PID check in place)

Reply:
Just to make sure we're on the same page: This assert _needs_ to be changed in the context of this commit, because with the rest we'll actually be creating jobs with pgid = 0.